airflow/dag_processing/processor.py

```diff
@@ -157,7 +156,7 @@ class DagFileParseRequest(BaseModel):
 
     file: str
     requests_fd: int
-    callback_requests: list[CallbackRequest] = Field(default_factory=list)
+    callback_requests: list[DagCallbackRequest | TaskCallbackRequest] = Field(default_factory=list)
```

Comment:
This should also have continued to work:

```suggestion
    callback_requests: list[CallbackRequest] = Field(default_factory=list)
```

Reply:
The same - from 3.10 only.

Reply:
Type hints should work because of [`eval_type_backport`](https://pypi.org/project/eval-type-backport/) -- that we install.

Reply:
Fixed.